ui/src/components/CardListItem.vue

```diff
@@ -1,21 +1,168 @@
+                label.push(this.locations[i].key);
+                value.push(this.locations[i].doc_count);
+            }
+
+            const ctx = document.getElementById('location-chart');
 
+            new Chart(ctx, {
+                type: 'horizontalBar',
+                data: {
+                    labels: label,
+                    datasets: [{
+                        label: 'Job numbers by location',
+                        data: value
+                    }]
+                },
+                options: {
+                    scales: {
+                        yAexs: [{
+                            ticks: {
+                                beginAtZero: true
+                            }
+                        }]
+                    }
+                }
+            });
+        }
+        else if (this.employmentTypes) {
+               for (let i = 0; i < this.employmentTypes.length; i++) {
+                label.push(this.employmentTypes[i].key);
+                value.push(this.employmentTypes[i].doc_count);
+            }
+
+            const ctx = document.getElementById('employmentType-chart');
+
+            new Chart(ctx, {
+                type: 'horizontalBar',
+                data: {
+                    labels: label,
+                    datasets: [{
+                        label: 'Job numbers by Employment Types',
+                        data: value
+                    }]
+                },
+                options: {
+                    scales: {
+                        yAexs: [{
+                            ticks: {
+                                beginAtZero: true
+                            }
+                        }]
+                    }
+                }
+            });
+        }
+        else if (this.skills) {
+               for (let i = 0; i < this.skills.length; i++) {
+                label.push(this.skills[i].key);
+                value.push(this.skills[i].doc_count);
+            }
+
+            const ctx = document.getElementById('skill-chart');
+
+            new Chart(ctx, {
+                type: 'horizontalBar',
+                data: {
+                    labels: label,
+                    datasets: [{
+                        label: 'Job numbers by Skills',
+                        data: value
+                    }]
+                },
+                options: {
+                    scales: {
+                        yAexs: [{
+                            ticks: {
+                                beginAtZero: true
+                            }
+                        }]
+                    }
+                }
+            });
+        }
+        else if (this.isRemote) {
+               for (let i = 0; i < this.isRemote.length; i++) {
+                label.push(this.isRemote[i].key);
+                value.push(this.isRemote[i].doc_count);
+            }
+
+            const ctx = document.getElementById('remote-chart');
+
+            new Chart(ctx, {
```

Comment:
Can we maybe use some colours for the bars? And remove the vertical lines for the ticks?

Reply:
Added function to generate a random colour for each bar in the graph. If you want to generate same colour in each load (whenever user load the page, show same colour as previous page if the result is same), then I will find another way to fill colours.

Reply:
? No updates happened on the pull request